activerecord/lib/active_record/connection_adapters/mysql/column.rb

```diff
@@ -5,7 +5,7 @@ class Column < ConnectionAdapters::Column # :nodoc:
         delegate :extra, to: :sql_type_metadata, allow_nil: true
 
         def unsigned?
-          /\bunsigned\z/.match?(sql_type)
+          !/\A(?:enum|set)\b/.match?(sql_type) && /\bunsigned\b/.match?(sql_type)
```

Comment:
`enum` and `set` types does not allow to be defined as `unsigned`. But if defined as `enum('foo','bar','unsigned')`, `column.unsigned?` returns `true`. It is incorrect.

https://github.com/rails/rails/blob/master/activerecord/test/cases/adapters/mysql2/enum_test.rb#L12-L15

```ruby
  def test_should_not_be_unsigned
    column = EnumTest.columns_hash["enum_column"]
    assert_not column.unsigned?
  end
```